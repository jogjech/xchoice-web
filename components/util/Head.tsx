import React from "react";
import NextHead from "next/head";
import GoogleFonts from "next-google-fonts";
import { FunctionComponent } from "react";

interface Props {
  title: string;
}

const Head: FunctionComponent<Props> = ({ title, children }) => (
  <>
    <GoogleFonts href="https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&family=Merriweather:wght@300;400&display=swap" />
    <NextHead>
      <meta charSet="UTF-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <meta httpEquiv="x-ua-compatible" content="ie=edge" />

      <title>{title}</title>

      {children}
    </NextHead>
  </>
);
export default Head;
