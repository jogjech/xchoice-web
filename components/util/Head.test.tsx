import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import Head from "./Head";

describe("Head util", () => {
  const title = "Title";
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Head title={title}></Head>);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains title tag", () => {
    const titleTag = wrapper.find("title");
    expect(titleTag.text()).toEqual(title);
  });
});
