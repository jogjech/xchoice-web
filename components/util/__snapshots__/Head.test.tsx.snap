// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head util renders correctly 1`] = `
<Fragment>
  <Component
    href="https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&family=Fredoka+One&family=Merriweather:wght@300;400&display=swap"
  />
  <Head>
    <meta
      charSet="UTF-8"
    />
    <meta
      content="width=device-width, initial-scale=1.0"
      name="viewport"
    />
    <meta
      content="ie=edge"
      httpEquiv="x-ua-compatible"
    />
    <title>
      Title
    </title>
  </Head>
</Fragment>
`;
